### ISSUE TITLE
ds8 503 pch******dla

### ISSUE THREAD

**Issue by jandersonlee** [data-8/jupyterhub-deploy#25]
See error logs:
[https://docs.google.com/document/d/1p-ayWS3LvYWGjwAHtO89ikPJzfQ4sW3FiduyUViSHVM/edit]

The container died on startup which caused the configurable http proxy to fail. 

`[W 2016-02-24 02:11:31.557 JupyterHub base:315] User pch******dla server stopped, with exit code: ExitCode=1, Error='', FinishedAt=2016-02-24T02:12:03.933397133Z`
`[I 2016-02-24 02:11:31.558 JupyterHub orm:194] Removing user pch******dla from proxy`

Even though it dies, a new route gets added:

`W 2016-02-24 02:12:17.829 JupyterHub orm:232] Adding missing route for pch******dla`
`[I 2016-02-24 02:12:17.831 JupyterHub orm:179] Adding user pch******dla to proxy /user/pch******dla => http://ds12.CS.Berkeley.EDU:***75`

**Comment by ryanlovett** [comment]
It looks like the user's container wasn't created properly:
```
ds8-jhub-host:~# sdocker logs jupyter-pch******dla
Creating user pcharagondla (1677)
/opt/conda/lib/python3.5/site-packages/IPython/paths.py:69: UserWarning: IPython parent '/home/pcharagondla' is not a writable location, using a temp directory.
  " using a temp directory.".format(parent))
Traceback (most recent call last):
  File "/usr/local/bin/jupyterhub-singleuser", line 231, in <module>
    main()
  File "/usr/local/bin/jupyterhub-singleuser", line 227, in main
    return SingleUserNotebookApp.launch_instance()
  File "/opt/conda/lib/python3.5/site-packages/jupyter_core/application.py", line 267, in launch_instance
    return super(JupyterApp, cls).launch_instance(argv=argv, **kwargs)
  File "/opt/conda/lib/python3.5/site-packages/traitlets/config/application.py", line 588, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-7>", line 2, in initialize
  File "/opt/conda/lib/python3.5/site-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/opt/conda/lib/python3.5/site-packages/notebook/notebookapp.py", line 1017, in initialize
    super(NotebookApp, self).initialize(argv)
  File "<decorator-gen-6>", line 2, in initialize
  File "/opt/conda/lib/python3.5/site-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/opt/conda/lib/python3.5/site-packages/jupyter_core/application.py", line 243, in initialize
    self.migrate_config()
  File "/opt/conda/lib/python3.5/site-packages/jupyter_core/application.py", line 169, in migrate_config
    migrate()
  File "/opt/conda/lib/python3.5/site-packages/jupyter_core/migrate.py", line 238, in migrate
    ensure_dir_exists(env['jupyter_config'])
  File "/opt/conda/lib/python3.5/site-packages/ipython_genutils/path.py", line 167, in ensure_dir_exists
    os.makedirs(path, mode=mode)
  File "/opt/conda/lib/python3.5/os.py", line 241, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/home/pch******dla/.jupyter'
Creating user pch******dla (1677)
useradd: user 'pch******dla' already exists
```
Apparently /home/ isn't writable because /home isn't mounted on ds12. Investigating...

**Comment by ryanlovett** [comment]
ds12 was the only one not mounted and `mount /home` was sufficient to fix it. I removed the containers on it since they were create improperly during this period, `docker ps -aq | xargs docker rm`

**Comment by jandersonlee** [comment]
Good to know that mounting /home fixed it, but I'm just as concerned that even though the container startup failed there was a route added to the proxy.

**Comment by jandersonlee** [comment]
@ryanlovett 

`[W 2016-02-24 02:11:31.557 JupyterHub base:315] User pch******dla server stopped, with exit code: ExitCode=1, Error='', FinishedAt=2016-02-24T02:12:03.933397133Z]`

What version of the jupyterhub code are we running? The current version I found on github has the "User %s server stopped, with exit code: %s" error message generated on line 318, not 315:

[https://github.com/jupyter/jupyterhub/blob/master/jupyterhub/handlers/base.py]

**Comment by ryanlovett** [comment]
We're [using](https://github.com/data-8/jupyterhub-deploy/blob/master/roles/jupyterhub/files/Dockerfile) 0.4.1. You can browse the source for that version at https://github.com/jupyter/jupyterhub/blob/0.4.1/jupyterhub/handlers/base.py.

**Comment by jandersonlee** [comment]
Supposedly we are using 0.4.1, but the error message seems to say otherwise:

[W 2016-02-24 02:11:31.557 JupyterHub base:315] User pch****dla server
stopped, with exit code: ExitCode=1, Error='',
FinishedAt=2016-02-24T02:12:03.933397133Z

In 0.4.1, this warning is generated at line 318, not 315:

[image: Inline image 1]
Sign me, Confused.

**Comment by ryanlovett** [comment]
That's on data8 and the provisioning for ds8. If you:
```
ds8-jhub-host:~# docker exec -it jupyterhub python -c 'import jupyterhub; print jupyterhub.__version__'
0.4.0.dev
```
We updated data8 to 0.4.1 on Feb. 7 to address a proxy redirection issue, but mostly so we could sit on something that was tagged. Before that we were just on :lastest which made it hard to debug. We haven't had to rebuild ds8's jupyterhub since then which is why that is still 0.4.0.dev.

**Comment by jandersonlee** [comment]
> We haven't had to rebuild ds8's jupyterhub since then which is why that is still 0.4.0.dev.

Ah. Great. Good to know.

**Comment by ryanlovett** [comment]
Fixed by jupyter/dockerspawner commit 186b41c.
